examples/RNOneSignal/App.js

```diff
@@ -8,484 +8,692 @@
-    console.log('Data: ', openResult.notification.payload.additionalData);
-    console.log('isActive: ', openResult.notification.isAppInFocus);
-    console.log('openResult: ', openResult);
-
-    this.setState({
-      jsonDebugText:
-        'OPENED: \n' + JSON.stringify(openResult.notification, null, 2),
-    });
-  }
-
-  onIds(device) {
-    console.log('Device info: ', device);
-    this.setState({
-        userId: device.userId
-    });
-  }
-
-  onInAppMessageClicked(actionResult) {
-    console.log('actionResult: ', actionResult);
-    this.setState({
-      jsonDebugText: 'CLICKED: \n' + JSON.stringify(actionResult, null, 2),
-    });
-  }
-
-  render() {
-    return (
-      <ScrollView style={styles.scrollView}>
-        <View style={styles.container}>
-          <View>
-            <Image style={styles.imageStyle} source={{uri: imageUri}} />
-          </View>
-          <Text style={styles.welcome}>Welcome to React Native!</Text>
-          <Text style={styles.instructions}>To get started, edit index.js</Text>
-          <Text style={styles.instructions}>
-            Double tap R on your keyboard to reload,{'\n'}
-            Shake or press menu button for dev menu
-          </Text>
-          <Text style={styles.jsonDebugLabelText}>
-            {this.state.jsonDebugText}
-          </Text>
-          <View style={{flexDirection: 'row', overflow: 'hidden'}}>
-            <View style={styles.buttonContainer}>
-              <Button
-                style={styles.button}
-                onPress={() => {
-                  OneSignal.getTags(tags => {
-                    console.log('Did get tags: ', tags);
-
-                    this.setState({
-                      jsonDebugText: JSON.stringify(tags, null, 2),
-                    });
-                  });
-                }}
-                title="Get Tags"
-                color={this.state.buttonColor}
-              />
+
+    constructor(properties) {
+        super(properties);
+
+        this.state = {
+            // OneSignal states
+            // Privacy Consent states
+            hasPrivacyConsent: false, // App starts without privacy consent
+            isPrivacyConsentLoading: requiresPrivacyConsent,
+
+            // Device states
+            userId: '',
+            pushToken: '',
+
+            // Subscription states
+            isSubscribed: false,
+            isSubscriptionLoading: false,
+
+            // External User Id states
+            externalUserId: '',
+            isExternalUserIdLoading: false,
+
+            // Email states
+            email: '',
+            isEmailLoading: false,
+
+            // In-App Messaging states
+            iam_paused: false,
+
+            // Add more states here...
+
+            // Demo App states
+            debugText: ''
+        };
+
+        // Log level logcat is 6 (VERBOSE) and log level alert is 0 (NONE)
+        OneSignal.setLogLevel(6, 0);
```

Comment:
maybe split into other function and call it from the constructor? constructor is getting big